### DISCUSSION TITLE
Control "Storage Default Mode" Through Automation

### DISCUSSION THREAD

**Discussion by Darkclad117** [binsentsu/home-assistant-solaredge-modbus#199] [Q&A] [answered]
Hi all,

I've been using this project for some time and now would like to start controlling my invertor. I've completed the setup and can read all the data and change the settings from the UI, however I'm stuck when trying to do this from an automation. 

How can I set the "select.solaredge_storage_default_mode" entity from an automation? I've tried using Call Service from the automation however I can't see the entity in the Automation UI. Do you have any advise please?

![Automation](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/104465467/e51a64c7-7b1d-4ec0-8f67-4916f77e6484)
![Dashboard](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/104465467/dee552d3-50db-44cd-9aee-bea4fc539d76)

Thank you to everyone who has contributed to this project, I'm very grateful!

**Comment by mpredfearn** [comment, marked as answer]
The entity is a "select" not an "input select" - I know, rather subtle! So the service you need to call is "select.select_option" then set your default mode.
![Screenshot_2023-09-18_09-17-05](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/12936891/5154c919-bfa8-496d-892b-f70b3379fd31)

**Comment by Kap3** [comment]
Have you lost( from rs485) power meter??
I my old system without battery and power meter the app is the same.
